segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/handler/PravegaRequestProcessor.java

```diff
@@ -947,11 +948,16 @@ private Void handleException(long requestId, String segment, long offset, String
             invokeSafely(connection::send, new SegmentRead(segment, offset, true, false, EMPTY_BYTE_BUFFER, requestId), failureHandler);
         } else if (u instanceof CancellationException) {
             log.info(requestId, "Closing connection {} while performing {} due to {}.",
-                     connection, operation, u.toString());
+                    connection, operation, u.toString());
             connection.close();
+        } else if (u instanceof TokenExpiredException) {
+            log.warn(requestId, "Token was found as expired during operation '{}'.", operation, u);
+            invokeSafely(connection::send, new AuthTokenCheckFailed(requestId, clientReplyStackTrace,
+                    AuthTokenCheckFailed.ERROR_CODE_TOKEN_EXPIRED), failureHandler);
         } else if (u instanceof TokenException) {
-            log.warn(requestId, "Token verification failed during '{}'.", operation);
-            invokeSafely(connection::send, new AuthTokenCheckFailed(requestId, clientReplyStackTrace), failureHandler);
+            log.warn(requestId, "Token exception encountered during operation '{}'.", operation, u);
```

Comment:
Think carefully if you need to log the full stack trace of just exception message (`u.toString()`).

Reply:
Since the TokenException is a generic one, it'd be useful to see the stacktrace in the logs to determine the cause. This exception should rarely occur if it occurs at all. `TokenExpiredException`, on the other hand, is a specific exception; So, for `TokenExpiredException` we only log the exception message.